unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarLOD/AvatarsLODController.cs

```diff
@@ -139,11 +135,23 @@ private bool IsInInvisibleDistance(float distance)
             return firstPersonCamera ? distance < AVATARS_INVISIBILITY_DISTANCE : distance < 0f; // < 0 is behind camera
         }
 
-        private (IAvatarLODController lodController, float distance)[] ComposeLODControllersSortedByDistance(IEnumerable<IAvatarLODController> lodControllers, Vector3 ownPlayerPosition)
+        private (IAvatarLODController lodController, float distance)[] ComposeLODControllersSortedByDistance(Dictionary<string, IAvatarLODController>.ValueCollection lodControllers, Vector3 ownPlayerPosition)
         {
-            (IAvatarLODController lodController, float distance)[] lodControllersWithDistance = lodControllers.Select(x => (x, DistanceToOwnPlayer(x.player, ownPlayerPosition))).ToArray();
-            Array.Sort(lodControllersWithDistance, (x, y) => x.distance.CompareTo(y.distance));
+            (IAvatarLODController lodController, float distance)[] lodControllersWithDistance = new (IAvatarLODController x, float)[lodControllers.Count];
```

Comment:
I would use a cached List here to avoid allocating memory every time.

Reply:
@AjimenezDCL, can you re-check if this the adjustment that you meant?

Reply:
Agree with this!